src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -93,8 +93,15 @@ static HRESULT _duplicateHandle(const HANDLE in, HANDLE& out) noexcept
 //   from the registered handler event function.
 HRESULT CTerminalHandoff::EstablishPtyHandoff(HANDLE in, HANDLE out, HANDLE signal, HANDLE ref, HANDLE server, HANDLE client) noexcept
 {
+    // Stash a local copy of _pfnHandoff before we stop listening.

```

Comment:
I wonder if there is a race here between the monarch turning on a listener at the same time on another thread as a previous request is being served out of the mta Com pool and turning it off. Maybe we should make methods on this class like startlistening, stoplistening, and/or this establishptyhandoff one use a synchronization primitive so multiple things can't happen to it at once...